salt/client/__init__.py

```diff
@@ -263,9 +270,6 @@ def run_job(
         '''
         arg = salt.utils.args.condition_input(arg, kwarg)
 
-        # Subscribe to all events and subscribe as early as possible
```

Comment:
As I recall, there was a race condition here if we didn't subscribe as early as we could. (I don't recall the exact details, though.) Are we confident that moving the subscribe to later on in the code-path won't reintroduce this? I assume that the caching logic is designed to prevent this, but I wanted to mention it.